metadata/src/test/java/org/apache/kafka/controller/QuorumControllerTest.java

```diff
@@ -482,6 +483,100 @@ public void testUncleanShutdownBroker() throws Throwable {
         }
     }
 
+    @Test
+    public void testUncleanShutdownElrDisabled() throws Exception {
+        List<Integer> allBrokers = Arrays.asList(1, 2, 3);
+        short replicationFactor = (short) allBrokers.size();
+        long sessionTimeoutMillis = 500;
+
+        try (
+            LocalLogManagerTestEnv logEnv = new LocalLogManagerTestEnv.Builder(1).
+                build();
+            QuorumControllerTestEnv controlEnv = new QuorumControllerTestEnv.Builder(logEnv)
+                .setControllerBuilderInitializer(controllerBuilder ->
+                    controllerBuilder.setFenceStaleBrokerIntervalNs(TimeUnit.SECONDS.toNanos(15)))
+                .setSessionTimeoutMillis(OptionalLong.of(sessionTimeoutMillis))
```

Comment:
This was a little interesting. We have a check in `ClusterControlManager.registerBroker` that checks whether we have a valid session before fencing requests with `DuplicateBrokerRegistrationException`.

That check _should_ require that a broker be fenced before it can re-register with a new epoch, but the check is too soft since it has a real-time component. If the check was instead that the broker is fenced before we allow re-registration, we would be safe.